From a6d1597fc432a4452f715558faea0e09aae56c9a Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 23 Jun 2014 20:01:11 -0400
Subject: [scsi] qla2xxx: Reduce the time we wait for a command to complete during SCSI error handling

Message-id: <1403553672-3844-6-git-send-email-cdupuis@redhat.com>
Patchwork-id: 84412
O-Subject: [PATCH RHEL 7.1 e-stor 5/6] qla2xxx: Reduce the time we wait for a command to complete during SCSI error handling.
Bugzilla: 1090378
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

From: Chad Dupuis <chad.dupuis@qlogic.com>

Bug 1090378 (https://bugzilla.redhat.com/show_bug.cgi?id=1090378)

scsi commit id 478c3b03cb1ead4be1b68345b38f3a2eba7494e7

Signed-off-by: Chad Dupuis <chad.dupuis@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index df2a8fc..24f5265 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -779,7 +779,7 @@ static int
 qla2x00_eh_wait_on_command(struct scsi_cmnd *cmd)
 {
 #define ABORT_POLLING_PERIOD 1000
-#define ABORT_WAIT_ITER  ((10 * 1000) / (ABORT_POLLING_PERIOD))
+#define ABORT_WAIT_ITER  ((2 * 1000) / (ABORT_POLLING_PERIOD))
  unsigned long wait_iter = ABORT_WAIT_ITER;
  scsi_qla_host_t *vha = shost_priv(cmd->device->host);
  struct qla_hw_data *ha = vha->hw;
-- 
1.7.1